### PR TITLE
WINDUPRULE-394: Spring boot version support rule

### PR DESCRIPTION
**Pull request by m-brophy**
depends on PR https://github.com/windup/windup/pull/1364

### CODE REVIEW

**Comment by mrizzi** [comment]
retest

File: rules-reviewed/springboot/springboot.rhamt.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <rule id="ocp-001-00001">
+            <!-- rule condition, when it could be fired -->
+            <when>
+                <dependency groupId="org.springframework.boot" artifactId="{*}" fromVersion="1.0.0.RELEASE" toVersion="2.1.1.RELEASE" />
+            </when>
+            <!-- rule operation, what to do if it is fired -->
+            <perform>
+                <hint title="ocp-001-00001 Unsupported version of SpringBoot" effort="1"
+                      category-id="potential">
+                    <message><![CDATA[ocp-001-00001 Unsupported version of SpringBoot ]]></message>
```
Comment (mrizzi) [inline rules-reviewed/springboot/springboot.rhamt.xml]:
```suggestion
                    <message>SpringBoot has to be updated to a supported version></message>
```

File: rules-reviewed/springboot/springboot.rhamt.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    <link href="https://access.redhat.com/articles/3348731" title="RHOAR SpringBoot Supported Configurations">link</link>
```
Comment (mrizzi) [inline rules-reviewed/springboot/springboot.rhamt.xml]:
```suggestion
                    <link href="https://access.redhat.com/articles/3349341" title="RHOAR Spring Boot Supported Configurations" />
                    <link href="https://access.redhat.com/articles/3348731" title="RHOAR Component Details Overview" />
```

File: rules-reviewed/springboot/tests/springboot.rhamt.test.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<ruletest id="ocp-001-test"
```
Comment (mrizzi) [inline rules-reviewed/springboot/tests/springboot.rhamt.test.xml]: Change the test file consistently with the changes applied to the `springboot.rhamt.xml` file

File: rules-reviewed/springboot/tests/springboot.rhamt.test.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+          xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <testDataPath>data</testDataPath>
+    <rulePath>../springboot.rhamt.xml</rulePath>
+    <ruleset>
+        <rules>
+            <rule id="ocp-001-00001-test">
+                <when>
+                    <not>
+                            <hint-exists message="Unsupported version of spring-boot-starter-logging"/>
```
Comment (mrizzi) [inline rules-reviewed/springboot/tests/springboot.rhamt.test.xml]: Add the `<iterable-filter>` check
```suggestion
                       <iterable-filter size="8">
                            <hint-exists message="Unsupported version of spring-boot-starter-logging"/>
                        </iterable-filter>
```

**Comment by m-brophy** [comment]
Changes applied

File: rules-reviewed/rhr/springboot/springboot.rhamt.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<ruleset id="springboot" xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            Verify the version of the Springboot application is compatible with those supported by Red Hat Runtimes
```
Comment (mrizzi) [inline rules-reviewed/rhr/springboot/springboot.rhamt.xml]:
```suggestion
            Verify the version of the Spring Boot framework is compatible with those supported by Red Hat Runtimes
```

File: rules-reviewed/rhr/springboot/springboot.rhamt.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    </metadata>
+    <rules>
+        <rule id="springboot-00001">
+            <!-- rule condition, when it could be fired -->
+            <when>
+                <dependency groupId="org.springframework.boot" artifactId="{*}" fromVersion="1.0.0.RELEASE" toVersion="2.1.1.RELEASE" />
+            </when>
+            <!-- rule operation, what to do if it is fired -->
+            <perform>
+                <hint title="springboot-00001 Unsupported version of SpringBoot" effort="1" category-id="potential">
```
Comment (mrizzi) [inline rules-reviewed/rhr/springboot/springboot.rhamt.xml]: No need to add the rule ID to the hint title
```suggestion
                <hint title="Unsupported version of Spring Boot" effort="1" category-id="potential">
```

File: rules-reviewed/rhr/springboot/springboot.rhamt.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                    <message>SpringBoot has to be updated to a supported version></message>
```
Comment (mrizzi) [inline rules-reviewed/rhr/springboot/springboot.rhamt.xml]:
```suggestion
                    <message>Spring Boot has to be updated to a version supported by Red Hat Runtimes</message>
```

**Comment by PhilipCattanach** [comment]
Successfully tested using applications with Spring 1.3.0 version artifacts

**Pull request merged**